File: IQ/Classes/clsInputType.vb
Imports dataAccess
Imports System.Runtime.Serialization

<DataContract()> Public Class clsInputType

    Property ID As Integer
    Property code As String
    Property name As String

    Public ReadOnly Property displayName(langauge As clsLanguage)
        Get
            Return Me.name & " (" & Me.code & ")"
        End Get
    End Property

    Public Sub New(id As Integer, code As String, name As String)

        Me.ID = id
        Me.code = code
        Me.name = name

        iq.InputTypes.Add(Me.ID, Me)
        iq.i_inputType_code.Add(Me.code, Me)

    End Sub



    Public Sub New(code As String, name As String)

        Dim sql$
        sql$ = "INSERT INTO [InputType] (code,name) values (" & da.SqlEncode(code) & "," & da.SqlEncode(name) & ");"

        Me.ID = da.DBExecutesql(sql$, True)
        Me.code = code
        Me.name = name

        iq.InputTypes.Add(Me.ID, Me)
        iq.i_inputType_code.Add(Me.code, Me)

    End Sub
End Class

File: IQ/Classes/clsInputType.cs
using dataAccess;
using System.Runtime.Serialization;


[DataContract()]
public class clsInputType
{

    public int ID { get; set; }
    public string code { get; set; }
    public string name { get; set; }

    public dynamic get_displayName(clsLanguage langauge)
    {
        return this.name + " (" + this.code + ")";
    }

    public clsInputType(int id, string code, string name)
    {

        this.ID = id;
        this.code = code;
        this.name = name;

        iq.InputTypes.Add(this.ID, this);
        iq.i_inputType_code.Add(this.code, this);

    }



    public clsInputType(string code, string name)
    {

        object sql = null;
        sql = "INSERT INTO [InputType] (code,name) values (" + da.SqlEncode(code) + "," + da.SqlEncode(name) + ");";

        this.ID = System.Convert.ToInt32(da.DBExecutesql(sql, true));
        this.code = code;
        this.name = name;

        iq.InputTypes.Add(this.ID, this);
        iq.i_inputType_code.Add(this.code, this);

    }
}